### ISSUE TITLE
Problem with Mlp module when fitting regression

### ISSUE THREAD

**Issue by PopovicMilica** [keras-team/autokeras#726]
A version of autokeras that I used: 0.4.0

I was trying to fit the Mlp module on the Boston dataset. Everything seemed to work fine, until I got to the prediction part, where all of my predicted targets were grouped at one value, with no variation. The code that I used is as follows:

```
import tensorflow as tf
import torch
#import numpy as np
#import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
from autokeras import MlpModule
from autokeras.backend import Backend
from autokeras.backend.torch import DataTransformerMlp
from autokeras.backend.torch.loss_function import regression_loss
from autokeras.backend.torch.metric import regression_metric
from abc import abstractmethod
import sklearn.datasets
from sklearn.model_selection import train_test_split

X, y = sklearn.datasets.load_boston(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=1)

class MSE:
    @classmethod
    def higher_better(cls):
        return False

    @classmethod
    def compute(cls, prediction, target):
        return Backend.regression_metric(prediction, target)

    @classmethod
    def evaluate(cls, prediction, target):
        return mean_squared_error(target, prediction)

mlpModule_untreat = MlpModule(loss = regression_loss, metric = MSE, searcher_args = {},
                              path = '~/.../Boston',
                              verbose=True)
data_transformer = DataTransformerMlp(X_train)
train_data = data_transformer.transform_train(X_train, y_train, batch_size = len(X_train))
test_data = data_transformer.transform_test(X_test, y_test,  batch_size = len(X_test))
fit_args = {
    "n_output_node": 1,
    "input_shape": X_train.shape,
    "train_data": train_data,
    "test_data": test_data
}
mlpModule_untreat.fit(n_output_node=fit_args.get("n_output_node"),
              input_shape=fit_args.get("input_shape"),
              train_data=fit_args.get("train_data"),
              test_data=fit_args.get("test_data"),
              time_limit=1 * 60 * 60)
```
<img width="359" alt="autokeras training" src="https://user-images.githubusercontent.com/29882027/62056971-4cc2b780-b1e4-11e9-8e23-4d17f38783bf.PNG">

```
# It trains 15 models in one hour. 
# Looking at the results
mlpModule_untreat.final_fit(train_loader, test_loader, trainer_args ={'max_no_improvement_num':1, 'max_iter_num':300}, retrain = True)
x_test_loader = data_transformer.transform_test(X_test, batch_size = len(X_test))
y_predicted = mlpModule_untreat.predict(x_test_loader)

# Plot y_predicted vs y_test
plt.plot(y_predicted,y_test,'k.',markersize = 4)
plt.plot(y_test,y_test,'r-',lw=1)
plt.xlabel('y_pred_',fontsize = 14)
plt.ylabel('y_test_',fontsize = 14)
plt.xlim(np.min(y_test),np.max(y_test))
plt.ylim(np.min(y_test),np.max(y_test))
plt.title('autokeras')
```

<img width="216" alt="y_pred vs y_test" src="https://user-images.githubusercontent.com/29882027/62058107-77ae0b00-b1e6-11e9-8b17-6b019ee6008b.PNG">

* Side note: I tried: mlpModule_untreat = MlpModule(loss = regression_loss, metric = regression_metric, searcher_args = {}, path = '~/.../Boston', verbose=True), but I got the following error when I was calling the fit function:
Exception occurred at train() : 'function' object has no attribute 'compute'

I am guessing that I am doing something wrong, can you please give me any suggestions? Thanks!
